lib/sycamore/sycamore/transforms/detr_partitioner.py

```diff
@@ -183,9 +199,14 @@ def __init__(self, model_name_or_path, device=None):
         ]
 
         self.device = device
+        self._model_name_or_path = model_name_or_path
 
-        self.processor = AutoImageProcessor.from_pretrained(model_name_or_path)
-        self.model = DeformableDetrForObjectDetection.from_pretrained(model_name_or_path).to(self._get_device())
+        self.use_model_server = os.environ.get("USE_MODEL_SERVER") == "1"
+        if self._get_device() != "cpu" or not self.use_model_server:
```

Comment:
what is the first predicate used for?

Reply:
This 'if' is used to reduce the number of model downloads. When the model is not run locally, the model does not need to be downloaded.

Reply:
If the machine running this has hardware acceleration, it will run the model locally and so will need to download the model.

Reply:
I've removed this in a new commit.